example-site/components/TableErrorValidation/TableWithValidation.tsx

```diff
@@ -0,0 +1,373 @@
+import { DataTable } from './components/DataTable';
+
+type MultiTablePageProps = {
+	tableRef?: RefObject<HTMLTableElement>;
+};
+
+const locationTextInputId = 'location-text-input';
+const dateOfAssessmentDatePickerId = 'date-picker';
+const tableHeadingId = 'table-heading';
+const tableId = 'data-table';
+
+const defaultAssessmentFiles = [
+	{
+		documentType: 'RMS Vehicle registration',
+		file: 'filename.pdf',
+		size: '88kb',
+		label: 'Upload',
+		buttonId: 'rms-vehicle-registration-upload-button',
+		error: false,
+	},
+	{
+		documentType: 'Operational Plan of Management',
+		file: 'filename.pdf',
+		size: '88kb',
+		label: 'Upload',
+		buttonId: 'operational-plan-of-management-upload-button',
+		error: false,
+	},
+	{
+		documentType: 'Vehicle build and layout plans',
+		size: '',
+		label: 'Upload',
+		buttonId: 'vehicle-build-and-layout-plans-upload-button',
+		error: false,
+	},
+	{
+		documentType: 'Food Safety Supervisor Certificate for Alex',
+		file: 'filename.pdf',
+		size: '88kb',
+		label: 'Upload',
+		buttonId: 'food-safety-supervisor-certificate-for-alex-upload-button',
+		error: false,
+	},
+	{
+		documentType: 'Suggested menu or list of foods being sold',
+		file: 'filename.pdf',
+		size: '88kb',
+		label: 'Upload',
+		buttonId: 'suggested-menu-or-list-of-foods-being-sold-upload-button',
+		error: false,
+	},
+];
+export type AssessmentFile = (typeof defaultAssessmentFiles)[number];
+
+const locationPageError = {
+	href: `#${locationTextInputId}`,
+	message: 'Location of assessment',
+	name: 'location' as const,
+};
+
+const datePageError = {
+	href: `#${dateOfAssessmentDatePickerId}`,
+	message: 'Date of assessment',
+	name: 'date' as const,
+};
+
+const filesPageError = {
+	href: `#section-alert`,
+	message: 'Assessment files',
+	name: 'files' as const,
+};
+
+type PageError =
+	| typeof locationPageError
+	| typeof datePageError
+	| typeof filesPageError;
+
+// const getPageErrors = ({name}: {name: PageErrors[number]['name']}) => {
+
+// }
+
+const sampleTableError = {
+	href: '#vehicle-build-and-layout-plans-upload-button',
+	message: 'Upload files',
+	name: 'files',
+};
+type SampleTableError = typeof sampleTableError;
+
+export const TableWithValidation = ({ tableRef }: MultiTablePageProps) => {
+	const tableSectionAlertRef = useRef<HTMLDivElement>(null);
+
+	// Input data
+	const [dateValue, setDateValue] = useState<Date>();
+	const [locationValue, setLocationValue] = useState('');
+
+	// Table data
+	const [assessmentFiles, setAssessmentFiles] = useState(
+		defaultAssessmentFiles
+	);
+	const [currentAssessmentFile, setCurrentAssessmentFile] =
+		useState<AssessmentFile>();
+
+	// Errors
+	const [pageErrors, setPageErrors] = useState<PageError[]>([]);
+	const [tableErrors, setTableErrors] = useState<SampleTableError[]>([]);
+
+	// Drawer state
+	const [isDrawerOpen, openDrawer, closeDrawer] = useTernaryState(false);
+
+	// Action handlers (includes some mocking)
+	const mockUploadFile = () => {
```

Comment:
Setting the page error when you upload one of the other files feels weird

Reply:
I think I'm with you.. It should only set that one row to error unless the "Save and continue" button is pressed?

- [ ] Row error only set on interacted-with row, not on other rows until "Save and continue"